### ISSUE TITLE
Links among markdown files don't work if upstream path element contains space

### ISSUE THREAD

**Issue by dschlaep** [doxygen/doxygen#9861]
**Describe the bug**
Doxygen does not resolve markdown links of the form `[here](MyMarkdownFile.md)`  for me if my project resides on a path where an upstream path element contains a space, e.g., "path with space/MyProject".

Warnings: (note I removed non-relevant parts of the path and inserted `...` instead)

> Generating docs for page md_doc_additional_doc_MyMarkdownFile...
> /.../path with space/MyProject/doc/additional_doc/MyMarkdownFile.md:7: warning: unable to resolve reference to '/.../path' for \ref command

and

> Generating index page...
> /.../path with space/MyProject/README.md:12: warning: unable to resolve reference to '.../path' for \ref command
> 

This results in the following html documentation
```
Additional information can be found /.../path with [space/MyProject/doc/additional_doc/MyMarkdownFile.md](file:///.../path%20with%20space/MyProject/doc/html/_my_markdown_file_8md.html) "here".
```

**Expected behavior**
When I copy the exact same project to a path without spaces, then the markdown links are correctly generated and doxygen does not issue warnings.

The documentation is working correctly; the corresponding line to the one cited above is
```
Additional information can be found [here](file:///.../path_without_space/MyProject/doc/html/md_doc_additional_doc__my_markdown_file.html).
```

**Screenshots**
NA

**To Reproduce**
Attached [debug_doxygen_paths.zip](https://github.com/doxygen/doxygen/files/10745210/debug_doxygen_paths.zip) contains a simple "Hello Word" program in C with some extra markdown files `README.md` and `MyMarkdownFile.md` which link to each other:

> - `demonstrate_path_issue.sh` 
> - MyProject/
>     - doc/
>         - `Doxyfile`
>         - additional_doc/
>             - `MyMarkdownFile.md` which contains the link `[markdown README](README.md)`
>         - `doc_open.sh`
>    - src/
>        - `HW.c`
>    - `README.md` which contains the link `[here](doc/additional_doc/MyMarkdownFile.md)`
>    - `makefile`
> 

The bash script `demonstrate_path_issue.sh`
- creates two folders `"path_without_space"` and `"path with space"`
- copies "`MyProject/"` into each created folder
- enters each copy, i.e., `"path_without_space/MyProject"` and `"path with space/MyProject"`
- runs doxygen from "MyProject" using "doc/Doxyfile" to create the documentation -- see warnings that I copied above
- opens the html documentation -- see html documentation line that I copied above

**Version**
I tested with doxygen 1.9.3 and 1.9.6 which I compiled myself via macports "doxygen" and "doxygen-devel".
I tested this on three different Apple computers: two with Intel chips and macOS Ventura 13.2.1; and one with M1 Max and macOS Monterey 12.6.3.

**Stack trace**
NA

**Additional context**
I was not able to find this exact error reported in an earlier issue; however, I found some issues that reported different problems with links among markdown files, e.g., #9412 and #9021 (link to markdown sections/labels); #7595 (link path includes `.name`); #7032 (link path starts with `..`); #9129 (link includes unicode)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9863

**Comment by doxygen** [comment]
@albert-github I think there is a simpler solution as implemented with https://github.com/doxygen/doxygen/commit/b8c00336322b2f285677b5ca3aff4a25be353341: instead of adding a new internal command, the `@ref` command can also be changed to support quotes for the first parameter (as an option). This also allows referring to a file with spaces, e.g.

`@ref "my file.md" "A file with a space in the name"`

**Comment by albert-github** [comment]
I've been thinking about that solution too, but I rejected it as I was not sure whether or not it might break some other code.

An alternative code update has been made in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
